portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -63,11 +81,55 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentEntity.setProperty(COMMENT_TEXT, commentText);
     commentEntity.setProperty(TIME_STAMP, timeStamp);
 
+    // Used for testing, see TODO
+    System.out.println(getUploadedFileUrl(request, /* formInputElementName= */ "image"));
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect(REDIRECT_URL);
   }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload an image file. */
```

Comment:
It is better to use Optional as it makes it explicit that nothing might get returned.

Reply:
https://docs.google.com/document/d/1Dq_mcrOdyfbEB_ovC9tQGui8DLopkES2vwEejckLom0/edit#heading=h.5rm21e874sp9

Reply:
fixed

Reply:
fixed